File: packages/core/src/engine/verbs/dedupe.ts
/*!
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project.
 */

import type ColumnTable from 'arquero/dist/types/table/column-table'
import type { DedupeArgs, TableStore } from '../../index.js'
import type { Step } from '../../types.js'

/**
 * Executes an arquero dedupe operation.
 * @param step
 * @param store
 * @returns
 */

export async function dedupe(
	step: Step,
	store: TableStore,
): Promise<ColumnTable> {
	const { input, args } = step
	const { columns } = args as DedupeArgs
	const inputTable = await store.table(input)

	if (columns !== undefined) {
		return inputTable.dedupe(columns)
	}

	return inputTable.dedupe()
}

File: python/data_wrangling_components/engine/verbs/dedupe.py
#
# Copyright (c) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project.
#

from dataclasses import dataclass

from data_wrangling_components.table_store import TableStore
from data_wrangling_components.types import InputColumnListArgs, Step


@dataclass
class DedupeArgs(InputColumnListArgs):
    pass


def dedupe(step: Step, store: TableStore):
    """Removes Duplicates from a table.

    :param step:
        Parameters to execute the operation.
        See :py:class:`~data_wrangling_components.engine.types.SetOperationArgs`.
    :type step: Step
    :param store:
        Table store that contains the inputs to be used in the execution.
    :type store: TableStore

    :return: new table with the result of the operation.
    """
    args = DedupeArgs(columns=step.args.get("columns", None))
    input_table = store.get(step.input)
    output = input_table.drop_duplicates(args.columns)
    return output
